### PR TITLE
Addressing PA-1335: First time rancher cluster login

### PR DESCRIPTION
**Pull request by snigdha-px**
**What this PR does / why we need it**:
1)Create and update rancher client for both source and destination cluster
2)Update RKEMap with the values for source and destination cluster (secret key, access key, token, endpoint)
**Which issue(s) this PR fixes** (optional)
Closes #PA-1335

**Special notes for your reviewer**:
log : https://jenkins.pwx.dev.purestorage.com/job/Users/job/Snigdha/job/RKE-firstloginnew/22/consoleFull
log with cleanup : https://jenkins.pwx.dev.purestorage.com/job/Users/job/Snigdha/job/RKE-firstloginnew/23/consoleFull
P.S: 
1)Will revert the value of defaultScheduler ="rke" in common.go and SCHEDULER="rke" -n deploy-ssh.go before merging
2)Once the PR is reviewed, will run all the test cases with this branch before merging

### CODE REVIEW

File: drivers/scheduler/rke/rke.go [hunk 1]
```diff
@@ -69,61 +61,63 @@ func (r *Rancher) Init(scheduleOpts scheduler.InitOptions) error {
...
+	gomega.Expect(rkeToken).NotTo(gomega.Equal(""),
+		"RKE_TOKEN Environment variable should not be empty")
 	rkeParameters.Token = rkeToken
 	rkeParameters.AccessKey = strings.Split(rkeToken, ":")[0]
 	rkeParameters.SecretKey = strings.Split(rkeToken, ":")[1]
 	return &rkeParameters, nil
 }
 
+// UpdateClient updates the rancher client based on the current cluster context
+func (r *Rancher) UpdateClient(clusterName string) error {
```
Comment (trenukarya-px) [inline drivers/scheduler/rke/rke.go:83]: Pls use "Rancher" string in function name as well.

Comment (snigdha-px) [reply drivers/scheduler/rke/rke.go:83]: done

File: drivers/scheduler/rke/rke.go [hunk 2]
```diff
@@ -69,61 +61,63 @@ func (r *Rancher) Init(scheduleOpts scheduler.InitOptions) error {
...
+	masterNodeName := node.GetMasterNodes()[0].Name
+	endpoint := "https://" + masterNodeName + "/v3"
+	if clusterName == "destination-config" {
+		rkeToken = os.Getenv("DESTINATION_RKE_TOKEN")
+		gomega.Expect(rkeToken).NotTo(gomega.Equal(""),
+			"DESTINATION_RKE_TOKEN Environment variable should not be empty")
+	} else if clusterName == "source-config" {
+		rkeToken = os.Getenv("SOURCE_RKE_TOKEN")
+		gomega.Expect(rkeToken).NotTo(gomega.Equal(""),
+			"SOURCE_RKE_TOKENEnvironment variable should not be empty")
```
Comment (kshithijiyer-px) [inline drivers/scheduler/rke/rke.go:96]: Add space here after `"SOURCE_RKE_TOKEN Environment`

**Review by kshithijiyer-px — CHANGES_REQUESTED** [review]
Miinor comment

File: drivers/scheduler/rke/rke.go (hunk 2 above)
Comment (snigdha-px) [reply drivers/scheduler/rke/rke.go]: done

**Comment by snigdha-px** [comment]
> Miinor comment

done

File: deployments/deploy-ssh.sh [hunk 3]
```diff
@@ -34,7 +34,7 @@ if [ -z "${SPEC_DIR}" ]; then
 fi
 
 if [ -z "${SCHEDULER}" ]; then
-    SCHEDULER="k8s"
+    SCHEDULER="rke"
```
Comment (kshithijiyer-px) [inline deployments/deploy-ssh.sh:37]: Remember to remove it before you merge.

File: drivers/scheduler/rke/rke.go [hunk 4]
```diff
@@ -264,7 +258,7 @@ func (r *Rancher) DeleteRancherProject(uid string) error {
 // SaveSchedulerLogsToFile gathers all scheduler logs into a file
 func (r *Rancher) SaveSchedulerLogsToFile(n node.Node, location string) error {
 	driver, _ := node.Get(r.K8s.NodeDriverName)
-	// requires 2>&1 since docker logs command send the logs to stdrr instead of sdout
+	// requires 2>&1 since docker logs command send the logs to stderr instead of sdtout
```
Comment (kshithijiyer-px) [inline drivers/scheduler/rke/rke.go:261]: This should be be stdout right?

File: tests/common.go [hunk 5]
```diff
@@ -275,7 +274,7 @@ const (
 
 const (
 	oneMegabytes                          = 1024 * 1024
-	defaultScheduler                      = "k8s"
+	defaultScheduler                      = "rke"
```
Comment (kshithijiyer-px) [inline tests/common.go:277]: Make sure you revert it.

**Review by kshithijiyer-px — CHANGES_REQUESTED** [review]
Minor changes

File: deployments/deploy-ssh.sh (hunk 3 above)
Comment (snigdha-px) [reply deployments/deploy-ssh.sh:37]: sure

File: tests/common.go (hunk 5 above)
Comment (snigdha-px) [reply tests/common.go:277]: sure

File: drivers/scheduler/rke/rke.go (hunk 4 above)
Comment (snigdha-px) [reply drivers/scheduler/rke/rke.go]: done

**Review by mkoppal-px — APPROVED** [review]
LGTM

**Comment by snigdha-px** [comment]
> LGTM

Full log (except for share testcases) after all the changes: https://aetos.pwx.purestorage.com/resultSet/testSetID/280968

**Review by trenukarya-px — APPROVED** [review]

**Pull request merged**